hardhat.config.js

```diff
@@ -35,6 +35,50 @@ task("compile", "Compile Colony contracts with pinned Token").setAction(async ()
   }
 });
 
+task("test", "Run tests").setAction(async () => {
+  const app = express();
+  const port = 8545;
+
+  app.use(bodyParser.json());
+  app.post("/", async function (req, res) {
+    try {
+      const response = await hre.network.provider.request(req.body);
+      res.send({ jsonrpc: "2.0", result: response, id: req.body.id });
+    } catch (error) {
+      const abiCoder = new ethers.utils.AbiCoder();
+      const decoded = abiCoder.decode(["string"], `0x${error.data.slice(10)}`);
+      res.send({
+        jsonrpc: "2.0",
+        error: { message: `Error: VM Exception while processing transaction: reverted with reason string '${decoded[0]}'` },
+        id: req.body.id,
+      });
+    }
+  });
+
+  if (hre.network.name === "hardhat") {
+    app.listen(port, function () {
+      console.log(`Exposing the provider on port ${port}!`);
+    });
+  }
+
+  const ganacheAccounts = { addresses: {}, private_keys: {} };
+  // eslint-disable-next-line no-restricted-syntax
+  for (const account of config.networks.hardhat.accounts) {
+    const { privateKey } = account;
+    const publicAddress = ethers.utils.computeAddress(privateKey);
+    ganacheAccounts.addresses[publicAddress] = publicAddress;
+    ganacheAccounts.private_keys[publicAddress] = privateKey;
+  }
+
+  fs.writeFileSync("ganache-accounts.json", JSON.stringify(ganacheAccounts, null, 2));
+
+  const nFails = await runSuper();
+  if (nFails > 0 && !process.env.CI && hre.network.name === "hardhat") {
```

Comment:
What do you think about adding an additional command-line argument (i.e. `stay-alive`) to toggle this behavior?

Otherwise I'm assuming people will have to manually kill the process once they're doing investigating.

Reply:
They will, but that was the case before as well and hasn't been an issue?

Reply:
What happens in this situation?

1. Run `npm run test:contracts`
2. Test fails, process keeps running
3. Edit some code
4. Run `npm run test:contracts`

Will the tests run again, or will we have to manually kill the process before re-running?

Reply:
Tested this locally, confirmed that the test process doesn't stop, but it can be manually killed with Ctl+C once inspection is done.